### DISCUSSION TITLE
Default home

### DISCUSSION THREAD

**Discussion by DebugOk** [EssentialsX/Essentials#4229] [Ideas and Feature Suggestions] [answered]
# The command
/defaulthome <homename>
/dhome <homename>

# What does it do?
It sets a default home that you'll teleport to when running /home without any arguments.

So instead of having to do `/home Mainbase` every time, you can just do
```
/defaulthome Mainbase
/home
```
And it'll teleport you to your name named Mainbase

# How is this useful?
It'll save players the hassle of having to write their home name every time after /home

**Comment by DebugOk** [comment]
I should've mentioned that I'm able to make this, but I need to know if it's actually worth it or not

**Comment by pop4959** [comment, marked as answer]
Currently the default home `home` is already set with `/sethome` (no args) and teleported to via `/home` (no args). Only when there is more than one home it becomes necessary to type out the home name.

I don't know if it would really make sense to add this since the majority of players probably wouldn't actually utilize it. It also prevents /home from listing the available homes (which imho is much more useful when you have a large number of them). That being said, I don't think it's a terrible idea, and if you can find a way to do it differently than proposed and avoid breaking current behavior I think it would be a welcome change.

**Reply by DebugOk** [reply]
Perhaps adding a `list ` argument to home could resolve that? I'm not really sure how it'd make the most sense. If someone doesn't have a default some set it should still display all the homes with `/home`.

Or, perhaps `/homes` could be used to list all the homes with a default one set.

Another idea could be to use `/defaulthome` or an alias to teleport instead, thus leaving the original home command unmodified, or using a config option/make it togglable by the player

**Reply by pop4959** [reply]
A list argument would conflict with the home named "list". Listing homes for other users also works differently (via the `username:home` syntax) and would be confusing. `/homes` is also already an alias of `/home` and would cause issues for people using that alias.

`/defaulthome` is an appropriate idea to retain compatibility with how users currently expect `/home` to function, but if the goal was to reduce the amount of typing necessary, it's already defeated the purpose.

To be quite honest, the more I think about it the less I think it really needs to change. It has tab-completion, so for most homes (even long ones) all that is needed is `/home <letter><TAB>`. One can also set a home that is simply a letter (e.g. `/home h`). As far as I'm concerned, the gain here is too minor to warrant breaking the existing behavior *or* adding something new just to save one or two keystrokes.

Perhaps something to put on the TODO for Essentials 3.x.

**Reply by DebugOk** [reply]
Alright, fair enough

**Reply by bdotsamir** [reply]
I've just had this same idea and so I came hunting in the feature requests before submitting a PR for it.

My idea:
* New command: `/defaulthome`. Sets a default home if no arguments are passed to `/home`
* Un-alias `/homes` -> `/home`
* Make `/homes` *list* the homes that are available
* `/home`:
  * If there are no arguments:
    * If default home, teleport them to it
    * If no default home, list homes
  * If the player only has one home, teleport them to it.
  * If the player has no homes, go to spawn or error out

This would be a breaking change, likely semver minor at a minimum. Not taking into account changes to the API, lang translations, etc. Let me know.
